--- /home/benjaminm/code/oops-bundle/quenchxx/src/quenchxx/FieldsIO/FieldsIOArome.cc.tmp.bak	2025-04-08 17:32:19.087926592 +0200
+++ /home/benjaminm/code/oops-bundle/quenchxx/src/quenchxx/FieldsIO/FieldsIOArome.cc	2025-04-07 12:24:33.647469243 +0200
@@ -20,7 +20,7 @@
 // -----------------------------------------------------------------------------
 
 void readArome(const Geometry & geom,
-               const oops::Variables & vars,
+               const varns::Variables & vars,
                const eckit::Configuration & config,
                atlas::FieldSet & fset) {
   oops::Log::trace() << "quenchxx::readArome starting" << std::endl;
@@ -41,7 +41,7 @@
   fset.clear();
 
   // Variables to copy / to read
-  oops::Variables varsToRead;
+  varns::Variables varsToRead;
   for (const auto & var : vars) {
     if (var.name() == "air_pressure" || var.name() == "air_pressure_half") {
       // Get surface pressure and retrieve air_pressure or air_pressure_half from ak/bk
