--- /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/FieldsIO/FieldsIOGmsh.h.tmp.bak	2025-05-20 12:47:08.802713152 +0200
+++ /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/FieldsIO/FieldsIOGmsh.h	2025-05-20 12:42:49.674410744 +0200
@@ -13,9 +13,8 @@
 #include "atlas/field.h"
 #include "atlas/output/Gmsh.h"
 
-#include "oops/base/Variables.h"
-
 #include "quenchxx/FieldsIO/FieldsIOBase.h"
+#include "quenchxx/VariablesSwitch.h"
 
 namespace quenchxx {
   class Geometry;
@@ -34,7 +33,7 @@
 
   // Read
   void read(const Geometry &,
-            const oops::Variables &,
+            const varns::Variables &,
             const eckit::Configuration &,
             atlas::FieldSet &) const override;
 
