--- /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/Covariance.h.tmp.bak	2025-01-18 06:56:55.445721471 +0100
+++ /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/Covariance.h	2025-01-18 04:26:13.994877519 +0100
@@ -19,6 +19,8 @@
 #include "oops/util/Printable.h"
 
 #include "quenchxx/Increment.h"
+#include "quenchxx/IncrModCtlVec.h"
+#include "quenchxx/VariablesSwitch.h"
 
 namespace quenchxx {
   class Geometry;
@@ -36,14 +38,24 @@
 
   // Constructor/destructor
   Covariance(const Geometry &,
-             const oops::Variables &,
+             const varns::Variables &,
              const eckit::Configuration &,
              const State &,
              const State &)
     {}
+  Covariance(const Geometry &,
+             const varns::Variables &,
+             const eckit::Configuration &,
+             const State &)
+    {}
   ~Covariance()
     {}
 
+  // Linearize
+  void linearize(const State &,
+                 const Geometry &)
+    {}
+
   // Multiply and inverse multiply
   void multiply(const Increment &,
                 Increment &) const
@@ -52,6 +64,14 @@
                        Increment &) const
     {throw eckit::NotImplemented(Here());}
 
+  // Square-root and adjoint
+  void multiplySqrt(const IncrModCtlVec &,
+                    Increment &) const
+    {throw eckit::NotImplemented(Here());}
+  void multiplySqrtTrans(const Increment &,
+                         IncrModCtlVec &) const
+    {throw eckit::NotImplemented(Here());}
+
   // Randomization
   void randomize(Increment & dxo) const
     {throw eckit::NotImplemented(Here());}
