--- /home/benjaminm/code/oops-bundle/quenchxx/src/quenchxx/Geometry.cc.tmp.bak	2025-04-08 17:32:19.026926214 +0200
+++ /home/benjaminm/code/oops-bundle/quenchxx/src/quenchxx/Geometry.cc	2025-04-08 17:27:40.737165145 +0200
@@ -30,6 +30,7 @@
 #include "oops/util/Logger.h"
 
 #include "quenchxx/Fields.h"
+#include "quenchxx/GeometryIterator.h"
 
 #define ERR(e, msg) {std::string s(nc_strerror(e)); throw eckit::Exception(s + ": " + msg, Here());}
 
@@ -148,6 +149,9 @@
     checkLonLat(*checkLonLatConf);
   }
 
+  // Setup iterator
+  setupIterator(config);
+
   // Print summary
   if (params.printSummary.value()) {
     this->print(oops::Log::info());
@@ -163,7 +167,8 @@
   partitioner_(other.partitioner_), mesh_(other.mesh_), groupIndex_(other.groupIndex_),
   levelsAreTopDown_(other.levelsAreTopDown_), modelData_(other.modelData_), alias_(other.alias_),
   latSouthToNorth_(other.latSouthToNorth_), interpolation_(other.interpolation_),
-  duplicatePoints_(other.duplicatePoints_) {
+  duplicatePoints_(other.duplicatePoints_), iteratorDimension_(other.iteratorDimension_),
+  nnodes_(other.nnodes_), nlevs_(other.nlevs_), vertCoordAvg_(other.vertCoordAvg_) {
   oops::Log::trace() << classname() << "::Geometry starting" << std::endl;
 
   // Copy function space
@@ -238,7 +243,7 @@
 
 // -----------------------------------------------------------------------------
 
-std::vector<size_t> Geometry::variableSizes(const oops::Variables & vars) const {
+std::vector<size_t> Geometry::variableSizes(const varns::Variables & vars) const {
   oops::Log::trace() << classname() << "::variableSizes starting" << std::endl;
 
   std::vector<size_t> sizes;
@@ -252,6 +257,18 @@
 
 // -----------------------------------------------------------------------------
 
+std::vector<size_t> Geometry::variableSizes(const std::vector<std::string> & varNames) const {
+  oops::Log::trace() << classname() << "::variableSizes starting" << std::endl;
+
+  // Create variables
+  const varns::Variables vars(varNames);
+
+  oops::Log::trace() << classname() << "::variableSizes done" << std::endl;
+  return variableSizes(vars);
+}
+
+// -----------------------------------------------------------------------------
+
 void Geometry::print(std::ostream & os) const {
   oops::Log::trace() << classname() << "::print starting" << std::endl;
 
@@ -375,7 +392,7 @@
     } else {
       // Get variable to read
       const std::string varName = vertCoordConf->getString("variable");
-      const oops::Variables vertCoordVars(std::vector<std::string>({varName}));
+      const varns::Variables vertCoordVars(std::vector<std::string>({varName}));
 
       // Add group index for this variable
       groupIndex_[varName] = group.index_;
@@ -657,7 +674,7 @@
   // Get variable to read
   const std::string lonName = checkLonLatConf.getString("longitude", "longitude");
   const std::string latName = checkLonLatConf.getString("latitude", "latitude");
-  const oops::Variables lonLatVars(std::vector<std::string>({lonName, latName}));
+  const varns::Variables lonLatVars(std::vector<std::string>({lonName, latName}));
 
   // Add new group to read coordinates
   groupData coordGroup;
@@ -700,5 +717,73 @@
 }
 
 // -----------------------------------------------------------------------------
+
+void Geometry::setupIterator(const eckit::Configuration & config) {
+  oops::Log::trace() << classname() << "::setupIterator starting" << std::endl;
+
+  // Iterator dimension
+  iteratorDimension_ = config.getInt("iterator dimension", 2);
+  ASSERT((iteratorDimension_ == 2) || (iteratorDimension_ == 3));
+
+  // First group vertical coordinate field
+  const auto vertCoord = groups_[0].vertCoord_;
+
+  // Domain size
+  nnodes_ = vertCoord.shape(0);
+  nlevs_ = vertCoord.shape(1);
+
+  // Get ghost view
+  const auto ghostView = atlas::array::make_view<int, 1>(functionSpace_.ghost());
+
+  // Averaged vertical coordinate
+  const auto vertCoordView = atlas::array::make_view<double, 2>(vertCoord);
+  for (atlas::idx_t jlevel = 0; jlevel < nlevs_; ++jlevel) {
+    // Initialization
+    double vertCoordAvg = 0.0;
+    double counter = 0.0;
+
+    // Accumulation
+    for (atlas::idx_t jnode = 0; jnode < nnodes_; ++jnode) {
+      if (ghostView(jnode) == 0) {
+        vertCoordAvg += vertCoordView(jnode, jlevel);
+        counter += 1.0;
+      }
+    }
+
+    // Communication
+    comm_.allReduceInPlace(vertCoordAvg, eckit::mpi::sum());
+    comm_.allReduceInPlace(counter, eckit::mpi::sum());
+
+    // Normalization
+    if (counter > 0.0) {
+      vertCoordAvg /= counter;
+    }
+
+    // Update profile
+    vertCoordAvg_.push_back(vertCoordAvg);
+  }
+
+  oops::Log::trace() << classname() << "::setupIterator done" << std::endl;
+}
+
+// -----------------------------------------------------------------------------
+
+GeometryIterator Geometry::begin() const {
+  return GeometryIterator(*this, 0, 0);
+}
+
+// -----------------------------------------------------------------------------
+
+GeometryIterator Geometry::end() const {
+  return GeometryIterator(*this, nnodes_, nlevs_);
+}
+
+// -----------------------------------------------------------------------------
+
+std::vector<double> Geometry::verticalCoord(std::string & vcUnits) const {
+  return vertCoordAvg_;
+}
+
+// -----------------------------------------------------------------------------
 
 }  // namespace quenchxx
