--- /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/Geometry.cc.tmp.bak	2025-04-02 10:55:30.968045331 +0200
+++ /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/Geometry.cc	2025-04-02 11:08:23.431326248 +0200
@@ -30,6 +30,7 @@
 #include "oops/util/Logger.h"
 
 #include "quenchxx/Fields.h"
+#include "quenchxx/GeometryIterator.h"
 
 #define ERR(e, msg) {std::string s(nc_strerror(e)); throw eckit::Exception(s + ": " + msg, Here());}
 
@@ -138,6 +139,36 @@
     checkLonLat(*checkLonLatConf);
   }
 
+  // Iterator dimension
+  iteratorDimension_ = config.getInt("iterator dimension", 2);
+  ASSERT((iteratorDimension_ == 2) || (iteratorDimension_ == 3));
+
+  // First group vertical coordinate field
+  const auto vertCoord = groups_[0].vertCoord_;
+
+  // Domain size
+  nnodes_ = vertCoord.shape(0);
+  nlevs_ = vertCoord.shape(1);
+
+  // Averaged vertical coordinate
+  const auto vertCoordView = atlas::array::make_view<double, 2>(vertCoord);
+  for (atlas::idx_t jlevel = 0; jlevel < nlevs_; ++jlevel) {
+    double vertCoordAvg = 0.0;
+    double counter = 0.0;
+    for (atlas::idx_t jnode = 0; jnode < nnodes_; ++jnode) {
+      if (ghostView(jnode) == 0) {
+        vertCoordAvg += vertCoordView(jnode, jlevel);
+        counter += 1.0;
+      }
+    }
+    comm.allReduceInPlace(vertCoordAvg, eckit::mpi::sum());
+    comm.allReduceInPlace(counter, eckit::mpi::sum());
+    if (counter > 0.0) {
+      vertCoordAvg /= counter;
+    }
+    vertCoordAvg_.push_back(vertCoordAvg);
+  }
+
   // Print summary
   if (params.printSummary.value()) {
     this->print(oops::Log::info());
@@ -153,7 +184,8 @@
   partitioner_(other.partitioner_), mesh_(other.mesh_), groupIndex_(other.groupIndex_),
   levelsAreTopDown_(other.levelsAreTopDown_), modelData_(other.modelData_), alias_(other.alias_),
   latSouthToNorth_(other.latSouthToNorth_), interpolation_(other.interpolation_),
-  duplicatePoints_(other.duplicatePoints_) {
+  duplicatePoints_(other.duplicatePoints_), iteratorDimension_(other.iteratorDimension_),
+  nnodes_(other.nnodes_), nlevs_(other.nlevs_), vertCoordAvg_(other.vertCoordAvg_) {
   oops::Log::trace() << classname() << "::Geometry starting" << std::endl;
 
   // Copy function space
@@ -228,7 +260,7 @@
 
 // -----------------------------------------------------------------------------
 
-std::vector<size_t> Geometry::variableSizes(const oops::Variables & vars) const {
+std::vector<size_t> Geometry::variableSizes(const varns::Variables & vars) const {
   oops::Log::trace() << classname() << "::variableSizes starting" << std::endl;
 
   std::vector<size_t> sizes;
@@ -242,6 +274,18 @@
 
 // -----------------------------------------------------------------------------
 
+std::vector<size_t> Geometry::variableSizes(const std::vector<std::string> & varNames) const {
+  oops::Log::trace() << classname() << "::variableSizes starting" << std::endl;
+
+  // Create variables
+  const varns::Variables vars(varNames);
+
+  oops::Log::trace() << classname() << "::variableSizes done" << std::endl;
+  return variableSizes(vars);
+}
+
+// -----------------------------------------------------------------------------
+
 void Geometry::print(std::ostream & os) const {
   oops::Log::trace() << classname() << "::print starting" << std::endl;
 
@@ -750,5 +794,23 @@
 }
 
 // -----------------------------------------------------------------------------
+
+GeometryIterator Geometry::begin() const {
+  return GeometryIterator(*this, 0, 0);
+}
+
+// -----------------------------------------------------------------------------
+
+GeometryIterator Geometry::end() const {
+  return GeometryIterator(*this, nnodes_, nlevs_);
+}
+
+// -----------------------------------------------------------------------------
+
+std::vector<double> Geometry::verticalCoord(std::string & vcUnits) const {
+  return vertCoordAvg_;
+}
+
+// -----------------------------------------------------------------------------
 
 }  // namespace quenchxx
