--- /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/Geometry.cc.tmp.bak	2025-04-01 13:53:54.391698318 +0200
+++ /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/Geometry.cc	2025-04-01 13:54:55.939031080 +0200
@@ -30,6 +30,7 @@
 #include "oops/util/Logger.h"
 
 #include "quenchxx/Fields.h"
+#include "quenchxx/GeometryIterator.h"
 
 #define ERR(e, msg) {std::string s(nc_strerror(e)); throw eckit::Exception(s + ": " + msg, Here());}
 
@@ -203,6 +204,36 @@
   comm_.allReduceInPlace(duplicatedPointsCount, eckit::mpi::sum());
   duplicatePoints_ = (duplicatedPointsCount > 0);
 
+  // Iterator dimension
+  iteratorDimension_ = config.getInt("iterator dimension", 2);
+  ASSERT((iteratorDimension_ == 2) || (iteratorDimension_ == 3));
+
+  // First group vertical coordinate field
+  const auto vertCoord = groups_[0].vertCoord_;
+
+  // Domain size
+  nnodes_ = vertCoord.shape(0);
+  nlevs_ = vertCoord.shape(1);
+
+  // Averaged vertical coordinate
+  const auto vertCoordView = atlas::array::make_view<double, 2>(vertCoord);
+  for (atlas::idx_t jlevel = 0; jlevel < nlevs_; ++jlevel) {
+    double vertCoordAvg = 0.0;
+    double counter = 0.0;
+    for (atlas::idx_t jnode = 0; jnode < nnodes_; ++jnode) {
+      if (ghostView(jnode) == 0) {
+        vertCoordAvg += vertCoordView(jnode, jlevel);
+        counter += 1.0;
+      }
+    }
+    comm.allReduceInPlace(vertCoordAvg, eckit::mpi::sum());
+    comm.allReduceInPlace(counter, eckit::mpi::sum());
+    if (counter > 0.0) {
+      vertCoordAvg /= counter;
+    }
+    vertCoordAvg_.push_back(vertCoordAvg);
+  }
+
   // GeometryData
   if (interpolation_.getString("interpolation type") == "unstructured") {
     geomData_.reset(new oops::GeometryData(functionSpace_, fields_, levelsAreTopDown_, comm_));
@@ -223,7 +254,8 @@
   partitioner_(other.partitioner_), mesh_(other.mesh_), groupIndex_(other.groupIndex_),
   levelsAreTopDown_(other.levelsAreTopDown_), modelData_(other.modelData_), alias_(other.alias_),
   latSouthToNorth_(other.latSouthToNorth_), interpolation_(other.interpolation_),
-  duplicatePoints_(other.duplicatePoints_) {
+  duplicatePoints_(other.duplicatePoints_), iteratorDimension_(other.iteratorDimension_),
+  nnodes_(other.nnodes_), nlevs_(other.nlevs_), vertCoordAvg_(other.vertCoordAvg_) {
   oops::Log::trace() << classname() << "::Geometry starting" << std::endl;
 
   // Copy function space
@@ -255,7 +287,7 @@
     // Define group
     groupData group;
 
-    // Copy number of levels
+    // Copy number of levelsche
     group.levels_ = other.groups_[groupIndex].levels_;
 
     // Copy corresponding level for 2D variables (first or last)
@@ -298,7 +330,7 @@
 
 // -----------------------------------------------------------------------------
 
-std::vector<size_t> Geometry::variableSizes(const oops::Variables & vars) const {
+std::vector<size_t> Geometry::variableSizes(const varns::Variables & vars) const {
   oops::Log::trace() << classname() << "::variableSizes starting" << std::endl;
 
   std::vector<size_t> sizes;
@@ -312,6 +344,18 @@
 
 // -----------------------------------------------------------------------------
 
+std::vector<size_t> Geometry::variableSizes(const std::vector<std::string> & varNames) const {
+  oops::Log::trace() << classname() << "::variableSizes starting" << std::endl;
+
+  // Create variables
+  const varns::Variables vars(varNames);
+
+  oops::Log::trace() << classname() << "::variableSizes done" << std::endl;
+  return variableSizes(vars);
+}
+
+// -----------------------------------------------------------------------------
+
 void Geometry::print(std::ostream & os) const {
   oops::Log::trace() << classname() << "::print starting" << std::endl;
 
@@ -365,7 +409,7 @@
   int ncid, retval, nlon_id, nlat_id, lon_id, lat_id, lsm_id;
 
   if (comm_.rank() == 0) {
-    // Open NetCDF file
+    // Open NetCDF fileche
     if ((retval = nc_open(maskPath.c_str(), NC_NOWRITE, &ncid))) ERR(retval, maskPath);
 
     // Get lon/lat sizes
@@ -403,7 +447,7 @@
       lon[ilon] = static_cast<double>(zlon[ilon]);
     }
     for (size_t ilat = 0; ilat < nlat; ++ilat) {
-      lat[ilat] = static_cast<double>(zlat[ilat]);
+      lat[ilat] = static_cast<double>(zlat[ilat]);che
     }
     for (size_t ilat = 0; ilat < nlat; ++ilat) {
      for (size_t ilon = 0; ilon < nlon; ++ilon) {
@@ -760,5 +804,23 @@
 }
 
 // -----------------------------------------------------------------------------
+
+GeometryIterator Geometry::begin() const {
+  return GeometryIterator(*this, 0, 0);
+}
+
+// -----------------------------------------------------------------------------
+
+GeometryIterator Geometry::end() const {
+  return GeometryIterator(*this, nnodes_, nlevs_);
+}
+
+// -----------------------------------------------------------------------------
+
+std::vector<double> Geometry::verticalCoord(std::string & vcUnits) const {
+  return vertCoordAvg_;
+}
+
+// -----------------------------------------------------------------------------
 
 }  // namespace quenchxx
