--- src/quenchxx/Geometry.h.tmp	2024-11-15 14:40:50.970885627 +0100
+++ src/quenchxx/Geometry.h	2024-11-15 09:10:32.665953161 +0100
@@ -33,6 +33,7 @@
 }
 
 namespace quenchxx {
+  class GeometryIterator;
 
 // -----------------------------------------------------------------------------
 /// Orography parameters
@@ -213,6 +214,17 @@
   bool duplicatePoints() const
     {return duplicatePoints_;}
 
+  // Geometry iterator
+  GeometryIterator begin() const;
+  GeometryIterator end() const;
+  std::vector<double> verticalCoord(std::string &) const;
+  const size_t & iteratorDimension() const
+    {return iteratorDimension_;}
+  const size_t & nnodes() const
+    {return nnodes_;}
+  const size_t & nlevs() const
+    {return nlevs_;}
+
  private:
   // Print
   void print(std::ostream &) const;
@@ -279,6 +291,12 @@
 
   // Duplicate points
   bool duplicatePoints_;
+
+  // Geometry iterator
+  size_t iteratorDimension_;
+  size_t nnodes_;
+  size_t nlevs_;
+  std::vector<double> vert_coord_avg_;
 };
 
 // -----------------------------------------------------------------------------
