--- /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/Geometry.h.tmp.bak	2025-05-20 14:17:31.023204862 +0200
+++ /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/Geometry.h	2025-05-20 11:20:44.100610767 +0200
@@ -21,7 +21,6 @@
 #include "eckit/mpi/Comm.h"
 
 #include "oops/base/GeometryData.h"
-#include "oops/base/Variables.h"
 #include "oops/mpi/mpi.h"
 #include "oops/util/ObjectCounter.h"
 #include "oops/util/parameters/OptionalParameter.h"
@@ -30,11 +29,14 @@
 #include "oops/util/parameters/RequiredParameter.h"
 #include "oops/util/Printable.h"
 
+#include "quenchxx/VariablesSwitch.h"
+
 namespace eckit {
   class Configuration;
 }
 
 namespace quenchxx {
+  class GeometryIterator;
 
 // -----------------------------------------------------------------------------
 /// Orography parameters
@@ -78,12 +80,9 @@
   // Orography
   oops::OptionalParameter<OrographyParameters> orography{"orography", this};
 
-  // Vertical coordinate
-  oops::OptionalParameter<std::vector<double>> vert_coord{"vert_coord", this};
-
-  // Vertical coordinate from file
-  oops::OptionalParameter<eckit::LocalConfiguration> vert_coordFromFile{"vert_coord from file",
-    this};
+  // Vertical coordinate configuration
+  oops::OptionalParameter<eckit::LocalConfiguration> vertCoordConf{
+    "vertical coordinate", this};
 
   // Mask type
   oops::Parameter<std::string> maskType{"mask type", "none", this};
@@ -148,12 +147,15 @@
   oops::Parameter<eckit::LocalConfiguration> modelData{"model data", eckit::LocalConfiguration(),
     this};
 
-  // Aliases for model files
+  // Variables name alias for model files
   oops::Parameter<std::vector<AliasParameters>> alias{"alias", {}, this};
 
   // Latitudes from south to north in files
   oops::Parameter<bool> latSouthToNorth{"latitude south to north", true, this};
 
+  // Check longitudes/latitudes from file
+  oops::OptionalParameter<eckit::LocalConfiguration> checkLonLat{"check lon/lat from file", this};
+
   // Interpolation parameters
   oops::OptionalParameter<InterpolationParameters> interpolation{"interpolation", this};
 };
@@ -173,7 +175,8 @@
   Geometry(const Geometry &);
 
   // Variables sizes
-  std::vector<size_t> variableSizes(const oops::Variables & vars) const;
+  std::vector<size_t> variableSizes(const varns::Variables & vars) const;
+  std::vector<size_t> variableSizes(const std::vector<std::string> &) const;
 
   // Levels direction
   bool levelsAreTopDown() const
@@ -199,11 +202,10 @@
   size_t levels(const size_t & groupIndex) const
     {return groups_[groupIndex].levels_;}
   size_t levels(const std::string & var) const
-    {return groups_[groupIndex_.at(var)].levels_;}
+    {return groups_[groupIndex(var)].levels_;}
   size_t groups() const
     {return groups_.size();}
-  size_t groupIndex(const std::string & var) const
-    {return groupIndex_.at(var);}
+  size_t groupIndex(const std::string & var) const;
   const eckit::LocalConfiguration & modelData() const
     {return modelData_;}
   const std::vector<eckit::LocalConfiguration> & alias() const
@@ -214,19 +216,27 @@
     {return interpolation_;}
   bool duplicatePoints() const
     {return duplicatePoints_;}
+  const eckit::mpi::Comm & timeComm() const
+    {return eckit::mpi::self();}
+  const atlas::Field & vertCoord(const std::string & var) const
+    {return groups_[groupIndex(var)].vertCoord_;}
+  const std::vector<double> & vertCoordAvg(const std::string & var) const
+    {return groups_[groupIndex(var)].vertCoordAvg_;}
   const oops::GeometryData & generic() const
     {return *geomData_;}
 
- private:
-  // Print
-  void print(std::ostream &) const;
-
-  // Read land-sea mask
-  void readSeaMask(const std::string &,
-                   const size_t &,
-                   const std::string &,
-                   atlas::Field &) const;
+  // Geometry iterator
+  GeometryIterator begin() const;
+  GeometryIterator end() const;
+  std::vector<double> verticalCoord(std::string &) const;
+  const size_t & iteratorDimension() const
+    {return iteratorDimension_;}
+  const size_t & nnodes() const
+    {return nnodes_;}
+  const size_t & nlevs() const
+    {return nlevs_;}
 
+ private:
   // Communicator
   const eckit::mpi::Comm & comm_;
 
@@ -253,10 +263,12 @@
 
   // Group data structure
   struct groupData {
+    GroupParameters params_;
+    size_t index_;
     size_t levels_;
     std::string lev2d_;
-    atlas::Field vert_coord_;
-    std::vector<double> vert_coord_avg_;
+    atlas::Field vertCoord_;
+    std::vector<double> vertCoordAvg_;
     double gmaskSize_;
   };
 
@@ -272,7 +284,7 @@
   // Model data configuration
   eckit::LocalConfiguration modelData_;
 
-  // Aliases
+  // Variables name alias
   std::vector<eckit::LocalConfiguration> alias_;
 
   // Latitudes from south to north in files
@@ -284,8 +296,34 @@
   // Duplicate points
   bool duplicatePoints_;
 
+  // Geometry iterator
+  size_t iteratorDimension_;
+  size_t nnodes_;
+  size_t nlevs_;
+  std::vector<double> vertCoordAvg_;
+
   // Geometry data structure
   std::unique_ptr<oops::GeometryData> geomData_;
+
+  // Private methods
+
+  // Print
+  void print(std::ostream &) const;
+
+  // Setup alias
+  void setupAlias(const GeometryParameters &);
+
+  // Setup group vertical coordinate
+  void setupVertCoord(groupData &);
+
+  // Setup group mask
+  void setupMask(groupData &);
+
+  // Check longitudes/latitudes from file
+  void checkLonLat(const eckit::Configuration &);
+
+  // Setup iterator
+  void setupIterator(const eckit::Configuration &);
 };
 
 // -----------------------------------------------------------------------------
