--- /home/benjaminm/code/oops-bundle/quenchxx/src/quenchxx/Increment.cc.tmp.bak	2024-11-27 13:23:55.586743866 +0100
+++ /home/benjaminm/code/oops-bundle/quenchxx/src/quenchxx/Increment.cc	2024-11-26 14:58:10.122932578 +0100
@@ -19,7 +19,21 @@
 // -----------------------------------------------------------------------------
 
 Increment::Increment(const Geometry & resol,
-                     const oops::Variables & vars,
+                     const varns::Variables & vars,
+                     const util::DateTime & vt)
+  : fields_(new Fields(resol, vars, vt)) {
+  oops::Log::trace() << classname() << "::Increment starting" << std::endl;
+
+  fields_->zero();
+
+  oops::Log::trace() << classname() << "::Increment done" << std::endl;
+}
+
+// -----------------------------------------------------------------------------
+
+Increment::Increment(const Geometry & resol,
+                     const varns::Variables & vars,
+                     const util::DateTime &,
                      const util::DateTime & vt)
   : fields_(new Fields(resol, vars, vt)) {
   oops::Log::trace() << classname() << "::Increment starting" << std::endl;
@@ -133,6 +147,85 @@
 }
 
 // -----------------------------------------------------------------------------
+
+eckit::Stream & operator<<(eckit::Stream & s,
+                           const Increment & dx) {
+  oops::Log::trace() << "Increment::operator<< starting" << std::endl;
+
+  s << dx.fields();
+
+  oops::Log::trace() << "Increment::operator<< done" << std::endl;
+  return s;
+}
+
+// -----------------------------------------------------------------------------
+
+eckit::Stream & operator>>(eckit::Stream & s,
+                           Increment & dx) {
+  oops::Log::trace() << "Increment::operator>> starting" << std::endl;
+
+  s >> dx.fields();
+
+  oops::Log::trace() << "Increment::operator>> done" << std::endl;
+  return s;
+}
+
+// -----------------------------------------------------------------------------
+
+oops::LocalIncrement Increment::getLocal(const GeometryIterator & geometryIterator) const {
+  int index = 0;
+  if (this->geometry()->iteratorDimension() == 2) {
+    std::vector<int> variableSizes;
+    for (const auto & var : this->variables()) {
+      variableSizes.push_back(static_cast<int>(var.getLevels()));
+    }
+    size_t valuesSize = std::accumulate(variableSizes.begin(), variableSizes.end(), 0);
+    std::vector<double> values(valuesSize);
+    for (const auto & var : this->variables()) {
+      const auto view = atlas::array::make_view<double, 2>(this->fields().fieldSet()[var.name()]);
+      for (size_t jlevel = 0; jlevel < var.getLevels(); ++jlevel) {
+        values[index] = view(geometryIterator.jnode(), jlevel);
+        ++index;
+      }
+    }
+    return oops::LocalIncrement(this->variables(), values, variableSizes);
+  } else {
+    std::vector<int> variableSizes(this->variables().size(), 1);
+    size_t valuesSize = this->variables().size();
+    std::vector<double> values(valuesSize);
+    for (const auto & var : this->variables()) {
+      const auto view = atlas::array::make_view<double, 2>(this->fields().fieldSet()[var.name()]);
+      values[index] = view(geometryIterator.jnode(), geometryIterator.jlevel());
+      ++index;
+    }
+    return oops::LocalIncrement(this->variables(), values, variableSizes);
+  }
+}
+
+// -----------------------------------------------------------------------------
+
+void Increment::setLocal(const oops::LocalIncrement & localIncrement,
+                         const GeometryIterator & geometryIterator) {
+  std::vector<double> values = localIncrement.getVals();
+  size_t index = 0;
+  if (this->geometry()->iteratorDimension() == 2) {
+    for (const auto & var : this->variables()) {
+      auto view = atlas::array::make_view<double, 2>(this->fields().fieldSet()[var.name()]);
+      for (size_t jlevel = 0; jlevel < var.getLevels(); ++jlevel) {
+        view(geometryIterator.jnode(), jlevel) = values[index];
+        ++index;
+      }
+    }
+  } else {
+    for (const auto & var : this->variables()) {
+      auto view = atlas::array::make_view<double, 2>(this->fields().fieldSet()[var.name()]);
+      view(geometryIterator.jnode(), geometryIterator.jlevel()) = values[index];
+      ++index;
+    }
+  }
+}
+
+// -----------------------------------------------------------------------------
 
 void Increment::print(std::ostream & os) const {
   oops::Log::trace() << classname() << "::print starting" << std::endl;
