--- /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/Interpolation.cc.tmp.bak	2025-01-25 06:30:40.428094338 +0100
+++ /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/Interpolation.cc	2025-01-25 06:31:32.782095816 +0100
@@ -13,6 +13,8 @@
 
 #include "oops/util/FieldSetHelpers.h"
 
+#include "quenchxx/VariablesSwitch.h"
+
 // -----------------------------------------------------------------------------
 
 namespace quenchxx {
@@ -80,7 +82,7 @@
     fset.haloExchange();
 
     // Apply unstructured interpolator
-    const oops::Variables vars(fset.field_names());
+    const varns::Variables vars(fset.field_names());
     std::vector<double> vals;
     unstructuredInterp_->apply(vars, fset, vals);
 
@@ -133,7 +135,7 @@
     }
 
     // Apply unstructured interpolator, adjoint
-    const oops::Variables vars(tgtFieldSet.field_names());
+    const varns::Variables vars(tgtFieldSet.field_names());
     unstructuredInterp_->applyAD(vars, srcFieldSet, vals);
 
     // Exchange FieldSet halo, adjoint
