--- /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/LinearVariableChange.cc.tmp.bak	2025-01-18 06:56:55.400721381 +0100
+++ /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/LinearVariableChange.cc	2025-01-18 06:22:21.133759114 +0100
@@ -37,8 +37,8 @@
     // Define output to input variables map
     ASSERT(params.inputVariables.value()->size() == params.outputVariables.value()->size());
     for (size_t jj = 0; jj < params.outputVariables.value()->size(); ++jj) {
-      map_[(*params.outputVariables.value())[jj].name()] =
-        (*params.inputVariables.value())[jj].name();
+      map_[(*params.outputVariables.value())[jj]] =
+        (*params.inputVariables.value())[jj];
     }
 
     // Read multiplicative factor
@@ -63,7 +63,7 @@
     util::readFieldSet(geom.getComm(),
                        geom.functionSpace(),
                        geom.variableSizes(*params.inputVariables.value()),
-                       params.inputVariables.value()->variables(),
+                       *params.inputVariables.value(),
                        conf,
                        multiplierFset_);
   }
@@ -74,7 +74,7 @@
 // -----------------------------------------------------------------------------
 
 void LinearVariableChange::changeVarTL(Increment & dx,
-                                       const oops::Variables & vars) const {
+                                       const varns::Variables & vars) const {
   oops::Log::trace() << classname() << "::changeVarTL starting" << std::endl;
 
   if (!multiplierFset_.empty()) {
@@ -106,7 +106,7 @@
 // -----------------------------------------------------------------------------
 
 void LinearVariableChange::changeVarInverseTL(Increment & dx,
-                                              const oops::Variables & vars) const {
+                                              const varns::Variables & vars) const {
   oops::Log::trace() << classname() << "::changeVarInverseTL starting" << std::endl;
 
   if (!multiplierFset_.empty()) {
@@ -138,7 +138,7 @@
 // -----------------------------------------------------------------------------
 
 void LinearVariableChange::changeVarAD(Increment & dx,
-                                       const oops::Variables & vars) const {
+                                       const varns::Variables & vars) const {
   oops::Log::trace() << classname() << "::changeVarAD starting" << std::endl;
 
   if (!multiplierFset_.empty()) {
@@ -170,7 +170,7 @@
 // -----------------------------------------------------------------------------
 
 void LinearVariableChange::changeVarInverseAD(Increment & dx,
-                                              const oops::Variables & vars) const {
+                                              const varns::Variables & vars) const {
   oops::Log::trace() << classname() << "::changeVarInverseAD starting" << std::endl;
 
   if (!multiplierFset_.empty()) {
