--- /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/LinearVariableChangeParameters.h.tmp	2025-01-18 06:56:55.552721686 +0100
+++ /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/LinearVariableChangeParameters.h	2025-01-18 06:31:04.953956098 +0100
@@ -8,10 +8,11 @@
 
 #pragma once
 
+#include <string>
+#include <vector>
+
 #include "eckit/config/LocalConfiguration.h"
 
-#include "oops/base/ParameterTraitsVariables.h"
-#include "oops/base/Variables.h"
 #include "oops/util/parameters/OptionalParameter.h"
 #include "oops/util/parameters/Parameters.h"
 
@@ -23,8 +24,8 @@
 class LinearVariableChangeParameters : public oops::Parameters {
   OOPS_CONCRETE_PARAMETERS(LinearVariableChangeParameters, Parameters)
  public:
-  oops::OptionalParameter<oops::Variables> inputVariables{"input variables", this};
-  oops::OptionalParameter<oops::Variables> outputVariables{"output variables", this};
+  oops::OptionalParameter<std::vector<std::string>> inputVariables{"input variables", this};
+  oops::OptionalParameter<std::vector<std::string>> outputVariables{"output variables", this};
   // ATLAS file (multiplicative factor)
   oops::OptionalParameter<eckit::LocalConfiguration> atlasFile{"atlas file", this};
 };
