--- /home/benjaminm/code/oops-bundle/quenchxx/src/quenchxx/LinearVariableChange.h.tmp.bak	2024-12-05 10:04:27.845768711 +0100
+++ /home/benjaminm/code/oops-bundle/quenchxx/src/quenchxx/LinearVariableChange.h	2024-12-05 10:07:07.761768365 +0100
@@ -21,6 +21,7 @@
 #include "quenchxx/Geometry.h"
 #include "quenchxx/Increment.h"
 #include "quenchxx/State.h"
+#include "quenchxx/VariablesSwitch.h"
 
 namespace quenchxx {
 
@@ -39,17 +40,17 @@
 
   // Linear variable changes: TL, inverseTL, AD and inverseAD
   void changeVarTL(Increment &,
-                   const oops::Variables &) const;
+                   const varns::Variables &) const;
   void changeVarInverseTL(Increment &,
-                          const oops::Variables &) const;
+                          const varns::Variables &) const;
   void changeVarAD(Increment &,
-                   const oops::Variables &) const;
+                   const varns::Variables &) const;
   void changeVarInverseAD(Increment &,
-                          const oops::Variables &) const;
+                          const varns::Variables &) const;
 
   // Trajectory setup
   void changeVarTraj(const State &,
-                     const oops::Variables &)
+                     const varns::Variables &)
     {}
 
  private:
