--- /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/State.cc.tmp.bak	2025-01-18 06:56:55.468721518 +0100
+++ /home/benjaminm/code/jedi-bundle/quenchxx/src/quenchxx/State.cc	2024-12-02 17:28:27.591362773 +0100
@@ -22,7 +22,7 @@
 // -----------------------------------------------------------------------------
 
 State::State(const Geometry & resol,
-             const oops::Variables & vars,
+             const varns::Variables & vars,
              const util::DateTime & vt)
   : fields_(new Fields(resol, vars, vt)) {
   oops::Log::trace() << classname() << "::State starting" << std::endl;
@@ -41,7 +41,7 @@
 
   const std::vector<std::string> varNames = file.has("state variables") ?
     file.getStringVector("state variables") : file.getStringVector("variables");
-  const oops::Variables vars(varNames);
+  const varns::Variables vars(varNames);
   fields_.reset(new Fields(resol, vars, util::DateTime()));
   if (file.has("filepath")) {
     oops::Log::info() << "Info     : Create state from file" << std::endl;
