--- /home/benjaminm/code/oops-bundle/quenchxx/src/quenchxx/VariableChange.h.tmp.bak	2024-12-05 10:04:27.688767689 +0100
+++ /home/benjaminm/code/oops-bundle/quenchxx/src/quenchxx/VariableChange.h	2024-12-05 10:04:42.968866717 +0100
@@ -11,6 +11,7 @@
 #include <memory>
 #include <ostream>
 #include <string>
+#include <vector>
 
 #include "eckit/config/Configuration.h"
 
@@ -18,6 +19,9 @@
 
 #include "quenchxx/Geometry.h"
 #include "quenchxx/State.h"
+#include "quenchxx/VariablesSwitch.h"
+
+#include "vader/vader.h"
 
 namespace quenchxx {
 
@@ -36,9 +40,9 @@
 
   // Variable changes: direct and inverse
   void changeVar(State &,
-                 const oops::Variables &) const;
+                 const varns::Variables &) const;
   void changeVarInverse(State &,
-                        const oops::Variables &) const;
+                        const varns::Variables &) const;
 
  private:
   // Print
@@ -47,6 +51,9 @@
 
   // Geometry reference
   const Geometry & geom_;
+
+  // VADER
+  std::unique_ptr<vader::Vader> vader_;
 };
 
 // -----------------------------------------------------------------------------
