--- /home/benjaminm/code/oops-bundle/quenchxx/src/quenchxx/VariableChangeParameters.h.tmp.bak	2024-12-03 10:42:07.762620182 +0100
+++ /home/benjaminm/code/oops-bundle/quenchxx/src/quenchxx/VariableChangeParameters.h	2024-12-03 10:36:22.304318645 +0100
@@ -8,6 +8,10 @@
 
 #pragma once
 
+#include <map>
+#include <string>
+#include <vector>
+
 #include "eckit/config/LocalConfiguration.h"
 
 #include "oops/base/VariableChangeParametersBase.h"
@@ -19,6 +23,11 @@
 
 class VariableChangeParameters : public oops::VariableChangeParametersBase {
   OOPS_CONCRETE_PARAMETERS(VariableChangeParameters, VariableChangeParametersBase)
+ public:
+  // VADER parameters
+  oops::Parameter<std::map<std::string, std::vector<std::string>>> vaderCustomCookbook{
+    "vader custom cookbook", vaderQuenchxxCustomCookbook(), this};
+  oops::OptionalParameter<vader::VaderParameters> vaderParam{"vader", this};
 };
 
 // -------------------------------------------------------------------------------------------------
